configure.py

```diff
@@ -364,7 +364,7 @@ def search_system_path(file_name: str) -> Optional[str]:  # type: ignore # Missi
               '-Wno-unused-parameter',
               '-fno-rtti',
               '-fno-exceptions',
-              '-std=c++11',
+              '-std=c++14',
```

Comment:
I am not opposed to upgrading to C++14 (would love being able to use `std::make_unique<>` for example), but that would probably be better done in a separate PR, with an update to `CONTRIBUTING.md` that specifies our current requirements. @jhasse what do you think?

Reply:
My only requirement is: CI shouldn't fail (so it has to compile on RHEL 8). So no opposition on my side, too :)